### PR TITLE
fix: update defaults and generate `sitemap.xml` in `dist` directory

### PR DESCRIPTION
**Pull request by ricardogobbosouza**
- [x] Update defaults options
- [x] Generate `sitemap.xml` directly on `dist` directory with the `generate:done` hook
- [x] Removed `generate` option

### CODE REVIEW

**Comment by ricardogobbosouza** [comment]
Hi @NicoPennec 
Could you take a look?

File: src/index.js [hunk 1]
```diff
@@ -9,39 +9,26 @@ const AsyncCache = require('async-cache')
...
-  generate: false,
-  exclude: [],
-  routes: [],
-  cacheTime: 1000 * 60 * 15,
-  filter: undefined,
-  gzip: false
-}
-
 module.exports = function module (moduleOptions) {
+  const defaults = {
```
Comment (manniL) [inline src/index.js]: Why moving `defaults` inside the fn?

File: src/index.js [hunk 2]
```diff
@@ -9,39 +9,26 @@ const AsyncCache = require('async-cache')
...
+    hostname: this.options.build.publicPath || undefined,
+    exclude: [],
+    routes: this.options.generate.routes || [],
+    cacheTime: 1000 * 60 * 15,
+    filter: undefined,
+    gzip: false
+  }
+
   const options = Object.assign({}, defaults, this.options.sitemap, moduleOptions)
+  options.pathGzip = (options.gzip) ? `${options.path}.gz` : options.path
```
Comment (manniL) [inline src/index.js]:
```suggestion
  options.pathGzip = options.gzip ? `${options.path}.gz` : options.path
```

File: src/index.js (hunk 1 above)
Comment (ricardogobbosouza) [reply src/index.js]: @manniL Moved in to use `this.options`

File: src/index.js (hunk 2 above)
Comment (ricardogobbosouza) [reply src/index.js]: @manniL Changed :smile:

File: src/index.js [hunk 3]
```diff
@@ -63,32 +50,29 @@ module.exports = function module (moduleOptions) {
...
-      // TODO on build process only
+    } else {
       // Save static routes
       fs.ensureDirSync(path.resolve(this.options.buildDir, 'dist'))
       fs.writeJsonSync(jsonStaticRoutesPath, staticRoutes)
+    }
+  })
 
-      // TODO on generate process only and not on build process
-      if (options.generate) {
```
Comment (manniL) [inline src/index.js]: Last thing I promise! 😄 

A notice for the user that `generate` isn't needed anymore

Comment (ricardogobbosouza) [reply src/index.js]: @manniL great review!
Added notice with `consola.warn`

File: src/index.js [hunk 4]
```diff
@@ -63,32 +51,33 @@ module.exports = function module (moduleOptions) {
...
-          await fs.ensureFile(xmlGeneratePath)
-          await fs.writeFile(xmlGeneratePath, xml)
-          if (options.gzip) {
-            const gzip = await sitemap.toGzip()
-            await fs.writeFile(gzipGeneratePath, gzip)
-          }
-        })()
-      }
+  if (options.generate) {
+    consola.warn('The option `generate` isn\'t needed anymore')
```
Comment (manniL) [inline src/index.js]:
```suggestion
    consola.warn('The option `sitemap.generate` isn\'t needed anymore')
```

Comment (manniL) [reply src/index.js]: So the user knows which prop it is exactly

Comment (ricardogobbosouza) [reply src/index.js]: Ready! :+1:

**Comment by manniL** [comment]
many thanks @ricardogobbosouza ☺️

**Comment by NicoPennec** [comment]
@ricardogobbosouza thank your for the nice feature, it's was in my ToDoList since a long time 👍 

@manniL thanks for the reviews 
But it's a breaking change from the current release.
The current release v0.2 is fully compliant with nuxt before hooks, so the commit message must be mention that point.

Apologize, I know that I'm very slow to review PRs currently (due to personal life), but please discuss with me before merge 😉

**Comment by manniL** [comment]
@NicoPennec Sorry! I tagged you for review some time ago 🙈 

Regarding the change: You are right! No release has been made yet though.
Do you prefer to do a major release after all the other changes and only support Nuxt >= 2.x or would you like to stick with Nuxt 1.x compat?

**Comment by ricardogobbosouza** [comment]
Resolves #27

**Comment by manniL** [comment]
@NicoPennec Short reminder that this was a breaking change (as you release v0.2.1 and v0.2.2)

**Pull request merged**